go-controller/pkg/node/default_node_network_controller.go

```diff
@@ -443,28 +444,99 @@ func handleNetdevResources(resourceName string) (string, error) {
 	return netdevice, nil
 }
 
-func createNodeManagementPorts(name string, nodeAnnotator kube.Annotator, waiter *startupWaiter,
-	subnets []*net.IPNet) ([]managementPortEntry, *managementPortConfig, error) {
-	// If netdevice name is not provided in the full mode then management port backed by OVS internal port.
-	// If it is provided then it is backed by VF or SF and need to determine its representor name to plug
-	// into OVS integrational bridge
+func exportManagementPortAnnotation(netdevName string, nodeAnnotator kube.Annotator) error {
+	klog.Infof("Exporting management port annotation")
+
+	deviceID, err := util.GetDeviceIDFromNetdevice(netdevName)
+	if err != nil {
+		return err
+	}
+	vfindex, err := util.GetSriovnetOps().GetVfIndexByPciAddress(deviceID)
+	if err != nil {
+		return err
+	}
+	pfindex, err := util.GetSriovnetOps().GetPfIndexByVfPciAddress(deviceID)
+	if err != nil {
+		return err
+	}
+
+	return util.SetNodeManagementPortAnnotation(nodeAnnotator, pfindex, vfindex)
+}
+
+func importManagementPortAnnotation(node *kapi.Node) (string, error) {
+	klog.Infof("Import management port annotation")
+	pfId, vfId, err := util.ParseNodeManagementPortAnnotation(node)
+
+	if err != nil {
+		return "", err
+	}
+
+	return util.GetSriovnetOps().GetVfRepresentorDPU(fmt.Sprintf("%d", pfId), fmt.Sprintf("%d", vfId))
+}
+
+func getManagementPortNetDev(netdevName string) (string, error) {
+	link, err := util.GetNetLinkOps().LinkByName(netdevName)
+	if err != nil {
+		if !util.GetNetLinkOps().IsLinkNotFoundError(err) {
+			return "", fmt.Errorf("failed to lookup %s link: %v", netdevName, err)
+		}
+		// this may not the first time invoked on the node after reboot
+		// netdev may have already been renamed to ovn-k8s-mp0.
+		link, err = util.GetNetLinkOps().LinkByName(types.K8sMgmtIntfName)
+		if err != nil {
+			return "", fmt.Errorf("failed to get link device for %s. %v", netdevName, err)
+		}
+	}
+	return link.Attrs().Name, err
```

Comment:
Might be good to log `link.Attrs().Name` vs. `netdevName` if netdevName != link.Attrs().Name for debugging purposes in the future.